draft-ietf-acme-acme.md

```diff
@@ -3052,6 +3053,27 @@ There are certain factors that arise in operational reality that operators of
 ACME-based CAs will need to keep in mind when configuring their services.
 For example:
 
+## Key Selection
+
+ACME relies on two different classes of key pair:
+
+* Account key pairs, which are used to authenticate account holders
+* Certificate key pairs, which are used to sign and verify CSRs (and whose
+  public keys are included in certificates)
+
+Of these two types of key pair, account key pairs are far more dangerous.  While
+the compromise of a certificate key pair allows the attacker to impersonate the
+entities named in the certificate for the lifetime of the certificate, the
+compromise of an account key pair allows the attacker to take full control of
+the victim's ACME account -- potentially causing unintended issuance or
+revocation.
+
+For this reason, it is RECOMMENDED that account key pairs be used for no other
+purpose besides ACME authentication.  For example, the public key of an account
+key pair SHOULD NOT be included in a certificate.  ACME clients and servers
+SHOULD verify that a CSR submitted in a finalize request does not contain a
+public key for any known account key pair.
```

Comment:
I get that the global lookup on keys might be costly, so you don't want to MUST this, but is there any reason not to put a MUST on the account key associated with the finalize request?

Reply:
Just for background: Typically, we have avoided placing MUSTs on servers in places where a server's local policy might reasonably override the MUST.  For example, for a long time, we didn't specify which keys a server MUST trust for revocation.  

In the revocation case, I think people agreed to the MUST because servers could just make their own custom revocation thing if they disagreed with the MUST.  Here, that doesn't really apply, but I think there's a real security requirement here.